Crash on terraform init with unexpected var

### Terraform Version

```shell
Terraform v1.15.2
on linux_amd64
```

### Terraform Configuration Files

```terraform
terraform {
  required_version = "~> 1.15"
}
```


### Debug Output

```
2026-05-11T14:49:09.447-0700 [INFO]  Terraform version: 1.15.2
2026-05-11T14:49:09.448-0700 [DEBUG] using github.com/hashicorp/go-tfe v1.94.0
2026-05-11T14:49:09.448-0700 [DEBUG] using github.com/hashicorp/hcl/v2 v2.24.0
2026-05-11T14:49:09.448-0700 [DEBUG] using github.com/hashicorp/terraform-svchost v0.2.1
2026-05-11T14:49:09.448-0700 [DEBUG] using github.com/zclconf/go-cty v1.18.0
2026-05-11T14:49:09.448-0700 [INFO]  Go runtime version: go1.25.8
2026-05-11T14:49:09.448-0700 [INFO]  CLI args: []string{"terraform", "init", "-var", "abc=def"}
2026-05-11T14:49:09.448-0700 [TRACE] Stdout is a terminal of width 146
2026-05-11T14:49:09.448-0700 [TRACE] Stderr is a terminal of width 146
2026-05-11T14:49:09.448-0700 [TRACE] Stdin is a terminal
2026-05-11T14:49:09.448-0700 [DEBUG] Attempting to open CLI config file: /home/alan/.terraformrc
2026-05-11T14:49:09.448-0700 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2026-05-11T14:49:09.448-0700 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2026-05-11T14:49:09.448-0700 [DEBUG] ignoring non-existing provider search directory /home/alan/.terraform.d/plugins
2026-05-11T14:49:09.448-0700 [DEBUG] ignoring non-existing provider search directory /home/alan/.local/share/terraform/plugins
2026-05-11T14:49:09.448-0700 [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2026-05-11T14:49:09.448-0700 [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2026-05-11T14:49:09.449-0700 [INFO]  CLI command args: []string{"init", "-var", "abc=def"}

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
Please report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

panic: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:26 +0x5e
github.com/hashicorp/terraform/internal/logging.PanicHandler()
        github.com/hashicorp/terraform/internal/logging/panic.go:84 +0x16a
panic({0x3de0620?, 0x72ed9d0?})
        runtime/panic.go:783 +0x132
github.com/hashicorp/terraform/internal/configs.(*Config).addProviderRequirements(0x0, 0xc0008aaf00, 0x1, 0x1)
        github.com/hashicorp/terraform/internal/configs/config.go:421 +0x3c
github.com/hashicorp/terraform/internal/configs.(*Config).ProviderRequirements(...)
        github.com/hashicorp/terraform/internal/configs/config.go:337
github.com/hashicorp/terraform/internal/command.(*InitCommand).getProvidersFromConfig(0xc0008a2000, {0x4deda30, 0xc00062a910}, 0x0, 0x0, {0x0, 0x0, 0x0}, {0x0, 0x0}, ...)
        github.com/hashicorp/terraform/internal/command/init.go:369 +0x1c5
github.com/hashicorp/terraform/internal/command.(*InitCommand).run(0xc0008a2000, 0xc0008820d0, {0x4dfc220, 0xc0005a2040})
        github.com/hashicorp/terraform/internal/command/init_run.go:213 +0x16b6
github.com/hashicorp/terraform/internal/command.(*InitCommand).Run(0xc0008a2000, {0xc000072120?, 0x0?, 0x0?})
        github.com/hashicorp/terraform/internal/command/init.go:74 +0x2ae
github.com/hashicorp/cli.(*CLI).Run(0xc000284280)
        github.com/hashicorp/cli@v1.1.7/cli.go:265 +0x4ed
main.realMain()
        github.com/hashicorp/terraform/main.go:337 +0x1df2
main.main()
        github.com/hashicorp/terraform/main.go:62 +0x13
```


### Expected Behavior

The extra var should be ignored or a nice warning/error provided, not a full crash.

### Actual Behavior

Terraform crash as shown in debug log

### Steps to Reproduce

1. Minimal empty module as above
2. `terraform init -var abc=def` where var.abc is not defined

### Additional Context

Tested on Terraform 1.14.9 and the issue was not present. Terraform init succeeds, and terraform plan shows an normal error due to the unexpected variable.

### References

_No response_

### Generative AI / LLM assisted development?

_No response_

--- a/.changes/v1.15/BUG FIXES-20260512-125801.yaml
+++ b/.changes/v1.15/BUG FIXES-20260512-125801.yaml
@@ -1,0 +1,5 @@
+kind: BUG FIXES
+body: Fix crash during provider installation when there is no config
+time: 2026-05-12T12:58:01.257167+02:00
+custom:
+    Issue: "38560"

--- a/internal/command/init.go
+++ b/internal/command/init.go
@@ -377,6 +377,10 @@ const (
 // dependency lock data based on the configuration.
 // The dependency lock file itself isn't updated here.
 func (c *InitCommand) getProvidersFromConfig(ctx context.Context, config *configs.Config, upgrade bool, pluginDirs []string, flagLockfile string, view views.Init) (output bool, resultingLocks *depsfile.Locks, safeInitAction SafeInitAction, authResult *getproviders.PackageAuthenticationResult, diags tfdiags.Diagnostics) {
+	if config == nil {
+		return false, nil, SafeInitActionInvalid, nil, diags
+	}
+
 	ctx, span := tracer.Start(ctx, "install providers from config")
 	defer span.End()
 

--- a/internal/command/init.go
+++ b/internal/command/init.go
@@ -378,7 +378,7 @@ const (
 // The dependency lock file itself isn't updated here.
 func (c *InitCommand) getProvidersFromConfig(ctx context.Context, config *configs.Config, upgrade bool, pluginDirs []string, flagLockfile string, view views.Init) (output bool, resultingLocks *depsfile.Locks, safeInitAction SafeInitAction, authResult *getproviders.PackageAuthenticationResult, diags tfdiags.Diagnostics) {
 	if config == nil {
-		return false, nil, SafeInitActionInvalid, nil, diags
+		return false, nil, SafeInitActionNotRelevant, nil, diags
 	}
 
 	ctx, span := tracer.Start(ctx, "install providers from config")

--- a/internal/command/init_test.go
+++ b/internal/command/init_test.go
@@ -6335,7 +6335,7 @@ func TestInit_configErrorsImpactingStateStore(t *testing.T) {
 	code := initCmd.Run(args)
 	testOutput := done(t)
 	if code != 1 {
-		t.Fatalf("expected apply to fail with code 1, got code %d: \n%s", code, testOutput.All())
+		t.Fatalf("expected init to fail with code 1, got code %d: \n%s", code, testOutput.All())
 	}
 	log.Printf("[TRACE] TestInit_configErrorsImpactingStateStore: init complete")
 	t.Logf("init output:\n%s", testOutput.Stdout())
@@ -6352,6 +6352,36 @@ func TestInit_configErrorsImpactingStateStore(t *testing.T) {
 		if !strings.Contains(cleanString(testOutput.Stderr()), e) {
 			t.Fatalf("unexpected error, expected %q, given: %s", e, testOutput.Stderr())
 		}
+	}
+}
+
+func TestInit_varValueWithoutConfig(t *testing.T) {
+	td := t.TempDir()
+	t.Chdir(td)
+	cfg1 := `terraform {}`
+
+	if err := os.WriteFile(filepath.Join(td, "main.tf"), []byte(cfg1), 0644); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	ui := cli.NewMockUi()
+	view, done := testView(t)
+	initCmd := &InitCommand{
+		Meta: Meta{
+			Ui:   ui,
+			View: view,
+		},
+	}
+
+	code := initCmd.Run([]string{"-var", "foo=bar"})
+	testOutput := done(t)
+	if code != 1 {
+		t.Fatalf("expected init to fail with code 1, got code %d: \n%s", code, testOutput.All())
+	}
+
+	expectedErr := "Error: Value for undeclared variable"
+	if !strings.Contains(cleanString(testOutput.Stderr()), expectedErr) {
+		t.Fatalf("unexpected error, expected %q, given: %s", expectedErr, testOutput.Stderr())
 	}
 }
 

